Java/LFU.java

```diff
@@ -0,0 +1,144 @@
+package lfu;
+import java.util.LinkedList;
+import java.util.ListIterator;
+import java.util.HashMap;
+
+/**
+ * @author salman
+ * @param <T>
+ */
+
+public class LFU<T> {
```

Comment:
Where did you read that Long names are not advisable ? Until 6 Words, you need not worry about long names. In my opinion, Java class name should be as descriptive as possible. If it's descriptive enough, it won't require any documentation.

In practice, I have personally written classes with names as long as 10 words. So it's fine.